packages/host/app/components/matrix/room.gts

```diff
@@ -110,6 +123,8 @@ export default class Room extends Component<Signature> {
   private roomResource = getRoom(this, () => this.args.roomId);
   private messagesToSend: TrackedMap<string, string | undefined> =
     new TrackedMap();
+  private lastMessagesSent: TrackedMap<string, MessageField | undefined> =
+    new TrackedMap();
```

Comment:
Why does a room need more than one "last message" sent?

Reply:
a room only has one last message, this is a map for `roomId` -> `lastMessage`. I followed `messagesToSend` map which is map from a room to a string message. Do you think we don't need a map for "last message"?

Reply:
Intuitively, it doesn't seem to me like we should need it. This component renders a room. A room has one last message. So what value is the map serving?

Reply:
This room component is used for 'past session' in ai panel component and it seems we don't create a new object of room if user move to another 'past session'. We only change the `roomId` argument and I think that is the reason why we use a map for `messagesToSend`. And that is also my consideration why I use a map for `lastMessage`. We need to know this `lastMessage` belongs to which room, so we are not showing pending message for a room that doesn't have `lastMessage`. CMIIW

Reply:
This is relying on the room component remaining rendered. If the goal is to make sure we remember `lastMessage` (and `messagesToSend` for that matter) even if the room is no longer rendered, we should store it in a map in a service, not in a component.